include/range/v3/view/any_view.hpp

```diff
@@ -304,43 +328,53 @@ namespace ranges
 
             template<typename Ref, category Cat>
             struct any_view_interface
+              : any_sentinel_interface
```

Comment:
The first pass I made at this didn't store a sentinel in the `any_view_impl`: `at_end` just passed the view to `ranges::end` to get a sentinel on-demand to compare with its iterator argument. This was great, except for the fact that `at_end` MUST be `const` and not all views have `end() const`. This is when I realized there's only ever one sentinel value for an `any_view`, so I could simply store it inside the view's type erasure. (And the obvious lemma for input views - there's only ever one iterator - occurred to me soon after.)

I like that the sentinel and view "share" a type erasure instead of being separately erased. It simplifies things, and avoids having another vtable pointer. Plus I can bring back the "don't store sentinels at all" approach as an optimization for views with `end() const`. As a matter of fact, I'll do that now.